src/qibo/transpiler/unitary_decompositions.py

```diff
@@ -61,22 +61,31 @@ def calculate_psi(unitary, magic_basis=magic_basis, backend=None):
             f"{backend.__class__.__name__} does not support `linalg.eig.`",
         )
 
-    magic_basis = backend.cast(magic_basis, dtype=magic_basis.dtype)
+    magic_basis = backend.cast(magic_basis)
+    unitary = backend.cast(unitary)
     # write unitary in magic basis
-    u_magic = np.transpose(np.conj(magic_basis)) @ unitary @ magic_basis
+    u_magic = (
+        backend.np.transpose(backend.np.conj(magic_basis), (1, 0))
+        @ unitary
+        @ magic_basis
+    )
     # construct and diagonalize UT_U
-    ut_u = np.transpose(u_magic) @ u_magic
+    ut_u = backend.np.transpose(u_magic, (1, 0)) @ u_magic
     # When the matrix given to np.linalg.eig is a diagonal matrix up to machine precision the decomposition
     # is not accurate anymore. decimals = 20 works for random 2q Clifford unitaries.
-    eigvals, psi_magic = np.linalg.eig(np.round(ut_u, decimals=20))
-    # orthogonalize eigenvectors in the case of degeneracy (Gram-Schmidt)
-    psi_magic, _ = np.linalg.qr(psi_magic)
+    if backend.__class__.__name__ == "TensorflowBackend":
```

Comment:
Is this because tensorflow's numpy doesn't have `linalg.eig`? If that's the case you could always add it in the backend `__init__`: `self.np.linalg.eig = np.linalg.eig`

Reply:
> Is this because tensorflow's numpy doesn't have `linalg.eig`? 

Yes.

> If that's the case you could always add it in the backend `__init__`: `self.np.linalg.eig = np.linalg.eig`

Problem is that `tensorflow`'s experimental `numpy` has no `linalg` module at all.